### PR TITLE
fix deleteRecord  

### PR DESCRIPTION
**Pull request by evazca**
The pr is to fix a bug about delete records. Currently sarama send the request to the controller.  It should be sent to the leader according the code in kafka client.

related with https://github.com/Shopify/sarama/issues/1377

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -81,6 +81,28 @@ func (err ConfigurationError) Error() string {
 // See https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-ErrorCodes
 type KError int16
 
+// MultiError is used to contain multi error.
+type MultiError struct {
```
Comment (varun06) [inline errors.go]: What's the reason to add this?

File: admin.go [hunk 2]
```diff
@@ -374,29 +374,50 @@ func (ca *clusterAdmin) DeleteRecords(topic string, partitionOffsets map[int32]i
...
-		Topics:  topics,
-		Timeout: ca.conf.Admin.Timeout,
-	}
-
-	b, err := ca.Controller()
-	if err != nil {
-		return err
+	partitionPerBroker := make(map[*Broker][]int32)
+	for partition := range partitionOffsets {
+		broker, err := ca.client.Leader(topic, partition)
```
Comment (varun06) [inline admin.go]: formatting looks strange, can you please run gofmt again?

**Comment by bai** [comment]
Thanks! What @varun06 said; apart from that could you please squash your commits 🙏

File: admin.go (hunk 2 above)
Comment (evazca) [reply admin.go]: gofmt again but nothing changed. Can you please tell me what's the format problem. I'll fix it manually.

File: errors.go (hunk 1 above)
Comment (evazca) [reply errors.go]: Currently the delete record request is sent to leader by leader. Thus we may receive multi error response and probably the errors are not the same.  I use MultiError to contain the array of error.

**Comment by evazca** [comment]
> Thanks! What @varun06 said; apart from that could you please squash your commits 🙏

I have rebased the commit & push again.  But I'm not sure whether it is ok to do it right now and shall I squash the commit again after the code review finished.

**Pull request merged**